compiler/rustc_session/src/parse.rs

```diff
@@ -217,6 +217,11 @@ pub struct ParseSess {
     pub proc_macro_quoted_spans: AppendOnlyVec<Span>,
     /// Used to generate new `AttrId`s. Every `AttrId` is unique.
     pub attr_id_generator: AttrIdGenerator,
+    /// Track parse errors that will cause unnecessary resolve errors.  This is a crude mechanism,
+    /// we should do something that is scope-based, instead of crate-global.
+    pub silence_resolve_errors: Lock<bool>,
+    /// Track parse errors that suggests changing an assignment to be an equality check.
+    pub silence_missing_comparison: Lock<Vec<Span>>,
```

Comment:
Similarly.

Reply:
This one will be trickier to write in a nice way :-/

Reply:
`.any(|(sp, _)| sp == span)`, or maybe use a `HashMap`?